diff --git a/cmake_modules/ThirdpartyToolchain.cmake b/cmake_modules/ThirdpartyToolchain.cmake
index e9ded4f..3efe1ed 100644
--- a/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cmake_modules/ThirdpartyToolchain.cmake
@@ -309,7 +309,8 @@ set(EP_COMMON_CMAKE_ARGS
     -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
     -DCMAKE_CXX_FLAGS=${EP_CXX_FLAGS}
     -DCMAKE_C_FLAGS=${EP_C_FLAGS}
-    -DCMAKE_INSTALL_LIBDIR=lib)
+    -DCMAKE_INSTALL_LIBDIR=lib
+    -DCMAKE_POLICY_VERSION_MINIMUM=3.5)
 
 macro(build_lucene)
     message(STATUS "Building lucene from source")
@@ -1106,7 +1107,7 @@ macro(build_arrow)
                         URL_HASH "SHA256=${PAIMON_ARROW_BUILD_SHA256_CHECKSUM}"
                         LOG_PATCH ON
                         PATCH_COMMAND ${CMAKE_COMMAND} -E chdir <SOURCE_DIR> bash -c
-                                      "[ -f .patched ] && echo '<SOURCE_DIR> patch already applied, ignore...' || patch -s -N -p1 -i '${PATCH_FILE}' && touch .patched"
+                                      "[ -f .patched ] && echo '<SOURCE_DIR> patch already applied, ignore...' || (patch -s -N -p1 -i '${PATCH_FILE}' && sed -i '/^set(EP_COMMON_CMAKE_ARGS/a\\    -DCMAKE_POLICY_VERSION_MINIMUM=3.5' cpp/cmake_modules/ThirdpartyToolchain.cmake && touch .patched)"
                         GIT_SUBMODULES "" GIT_SUBMODULES_RECURSE FALSE ${ARROW_CONFIGURE}
                         UPDATE_DISCONNECTED 1
                         BUILD_BYPRODUCTS "${ARROW_STATIC_LIB}"
